map.js

```diff
@@ -46,51 +46,90 @@ legend.onAdd = function() {
 };
 legend.addTo(map);
 
+/*
+fetch (states10mSource)
+  .then(response => response.json())
+  .then(function(json) {
+      L.vectorGrid.slicer(json, {
+        rendererFactory: L.canvas.tile,
+        vectorTileLayerStyles: {
+          sliced: function (properties, zoom) {
+            return stateStyle(properties);
+          }
+        }
+      }).addTo(map);
+  });
+*/
 const statesLayer = new ZoomShowHide().addTo(map);
 
 const ZOOM_THRESHOLD = 5;
 
-const states50m = L.geoJSON(null, {
-  style: stateStyle
-});
-statesLayer.addLayer(states50m);
+let states50m = '';
+
+//  statesLayer.addLayer(states50m);
+
+let states10m = '';
 
-const states10m = L.geoJSON(null, {
-  style: stateStyle
-});
-states10m.min_zoom = ZOOM_THRESHOLD + 1;
-statesLayer.addLayer(states10m);
+//  states10m.min_zoom = ZOOM_THRESHOLD + 1;
+//  tatesLayer.addLayer(states10m);
 
+/*
 function loadTopoToLayer(topology, layer) {
   const geojson = topojson.feature(topology, topology.objects.countries);
-  layer.addData(geojson);
+  layer.options.dataSource = geojson;
   addLayerInteraction(layer);
 }
+*/
 
 fetch(states50mSource)
   .then(response => response.json())
-  .then(topology => loadTopoToLayer(topology, states50m));
+  .then(function(json) {
+      states50m = L.vectorGrid.slicer(json, {
+        rendererFactory: L.canvas.tile,
+        vectorTileLayerStyles: {
+          countries: function(properties) {
+            return stateStyle(properties);
+          }
+        }
+      }).addTo(map);
```

Comment:
You are adding the layer directly to the map, not to `statesLayer`. The `.min_zoom` property is also no longer specified for the layers, so the zoom restriction wouldn't work anyway.

Reply:
layers are now added to the statesLayer. Zooming issue is fixed.